service/src/test/java/bio/terra/workspace/app/configuration/external/controller/ControlledGcpResourceApiControllerConnectedTest.java

```diff
@@ -0,0 +1,257 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.UUID;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MockMvc;
+
+/** Use this instead of ControlledGcpResourceApiTest, if you want to talk to real GCP. */
+public class ControlledGcpResourceApiControllerConnectedTest extends BaseConnectedTest {
+  private static final Logger logger =
+      LoggerFactory.getLogger(ControlledGcpResourceApiControllerConnectedTest.class);
+
+  @Autowired MockMvc mockMvc;
+  @Autowired MockMvcUtils mockMvcUtils;
+  @Autowired ObjectMapper objectMapper;
+  @Autowired UserAccessUtils userAccessUtils;
+
+  private ApiWorkspaceDescription workspace;
+  private ApiCreatedControlledGcpGcsBucket originalControlledBucket;
+
+  @BeforeEach
+  public void setup() throws Exception {
+    workspace =
+        mockMvcUtils.createWorkspaceAndCloudContext(userAccessUtils.defaultUserAuthRequest());
+    originalControlledBucket = createControlledGcsBucket();
+  }
+
+  @AfterEach
+  public void cleanup() throws Exception {
+    mockMvcUtils.deleteWorkspace(userAccessUtils.defaultUserAuthRequest(), workspace.getId());
+  }
+
+  @Test
+  public void cloneControlledGcsBucket_copyNothing() throws Exception {
+    ApiCloneControlledGcpGcsBucketResult cloneResult =
+        cloneControlledGcsBucket(
+            /*sourceWorkspaceId=*/ workspace.getId(),
+            originalControlledBucket.getResourceId(),
+            /*destWorkspaceId=*/ workspace.getId(),
+            ApiCloningInstructionsEnum.NOTHING);
+
+    // Assert clone result has no CreatedControlledGcpGcsBucket
+    assertNull(cloneResult.getBucket().getBucket());
+  }
+
+  @Test
+  public void cloneControlledGcsBucket_copyDefinition() throws Exception {
+    ApiCloneControlledGcpGcsBucketResult cloneResult =
+        cloneControlledGcsBucket(
+            /*sourceWorkspaceId=*/ workspace.getId(),
+            originalControlledBucket.getResourceId(),
+            /*destWorkspaceId=*/ workspace.getId(),
+            ApiCloningInstructionsEnum.DEFINITION);
+
+    // Assert bucket in clone result
+    ApiGcpGcsBucketResource cloneResultBucket = cloneResult.getBucket().getBucket().getGcpBucket();
+    String cloneBucketName = cloneResultBucket.getAttributes().getBucketName();
+    assertBucket(cloneResultBucket, ApiStewardshipType.CONTROLLED, cloneBucketName);
+
+    // Assert bucket returned by calling ControlledGcpResource.getBucket
+    ApiGcpGcsBucketResource gotBucket =
+        getControlledGcsBucket(workspace.getId(), cloneResultBucket.getMetadata().getResourceId());
+    assertBucket(gotBucket, ApiStewardshipType.CONTROLLED, cloneBucketName);
+  }
+
+  // TODO(PF-1930): Implement feature and uncomment test
```

Comment:
you can just annotate with disable with the bug as the description. https://mkyong.com/junit5/junit-5-how-to-disable-tests/